### PR TITLE
Reduce size of buffer, stringBuffer and tape.

### PR DESCRIPTION
**Pull request by ZhaiMo15**
In class JsonValue, the default size of buffer and stringBuffer, as well as long[] tape in class Tape, is 34M.
But in practice it's not necessary.
This patch reduce the size of them from 34M to its actual size.

### CODE REVIEW

**Comment by ZhaiMo15** [comment]
I believe in future, the JsonValue might be deep copied, thus the size of byte[] and long[] is important.

File: src/main/java/org/simdjson/SimdJsonParser.java [hunk 1]
```diff
@@ -34,7 +34,8 @@ public JsonValue parse(byte[] buffer, int len) {
     }
 
     private byte[] padIfNeeded(byte[] buffer, int len) {
-        if (buffer.length - len < PADDING) {
+        if (buffer.length - len < PADDING && len < capacity) {
+            byte[] paddedBuffer = new byte[len + PADDING];
```
Comment (piotrrzysko) [inline src/main/java/org/simdjson/SimdJsonParser.java:38]: The reason for maintaining the `paddedBuffer` all the time, regardless of whether it is necessary or not, is to avoid allocations on hot paths. However, I see at least two issues with padding in general. Firstly, it requires adding this extra branch. Secondly, it complicates the API: on one hand, the user doesn't need to be aware of it, but on the other hand, if they want to achieve the best performance, they should pad the input. Therefore, I've been considering removing the need for padding altogether. It should be possible, although I haven't thoroughly researched this topic.

To summarize: I'd start by verifying if removing the padding is possible. If so, I'd remove it and test the performance of the parser. If there is no regression compared to the current version with padding, we have a win-win situation.

File: src/main/java/org/simdjson/TapeBuilder.java [hunk 2]
```diff
@@ -206,7 +206,13 @@ void reset() {
     }
 
     JsonValue createJsonValue(byte[] buffer) {
-        return new JsonValue(tape, 1, stringBuffer, buffer);
+        Tape newTape = new Tape(tape);
+
+        int stringBufferLen = stringParser.getStringBufferIdx();
+        byte[] newStringBuffer = new byte[stringBufferLen];
+        System.arraycopy(stringBuffer, 0, newStringBuffer, 0, stringBufferLen);
```
Comment (piotrrzysko) [inline src/main/java/org/simdjson/TapeBuilder.java:213]: Is this change related to https://github.com/simdjson/simdjson-java/issues/36? I'm asking because I'm a bit concerned that we need another allocation on the parsing path.

File: src/main/java/org/simdjson/SimdJsonParser.java (hunk 1 above)
Comment (ZhaiMo15) [reply src/main/java/org/simdjson/SimdJsonParser.java:38]: I think by design, the padding is 64 bytes. However, the 'paddedBuffer' is 34MB, that's such a waste. I'm just change the padding size to 64 bytes.

File: src/main/java/org/simdjson/TapeBuilder.java (hunk 2 above)
Comment (ZhaiMo15) [reply src/main/java/org/simdjson/TapeBuilder.java:213]: And as I mentioned above, the default size of buffer and stringBuffer, as well as long[] tape in class Tape, is 34M. If we allocated 34M * 3 for each element, the cost is way to much.

File: src/main/java/org/simdjson/SimdJsonParser.java (hunk 1 above)
Comment (piotrrzysko) [reply src/main/java/org/simdjson/SimdJsonParser.java:38]: I agree with you that this is a waste. However, in your approach, you are potentially allocating a new array on every call of the `parse` method, which can be costly. 

I've been working on removing the padding entirely. It's a bit complicated, but we will see if it is feasible. I'll report back.
